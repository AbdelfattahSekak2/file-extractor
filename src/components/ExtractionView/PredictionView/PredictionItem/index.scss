@import "../../../../styles/mixins";

.prediction-item {
  display: flex;
  flex-direction: row;
  border-radius: $border-radius;
  margin-top: 30px;
  box-shadow: $box-shadow;
  label {
    color: $dark-grey;
    background: white;
    padding: 2px 10px;
    border-radius: $border-radius;
    width: 130px;
    font-size: 15px;
    font-weight: 500;
    line-height: 40px;
  }
  .details {
    position: relative;
    width: 100%;
    height: 100%;
    align-self: center;
    z-index: 2;
    background: #bbbaba;
    border-radius: 0 $border-radius $border-radius 0;
    .bar {
      z-index: -1;
      position: absolute;
      top: 0;
      background: $mindee-red;
      width: 100%;
      height: 100%;
    }
    p {
      padding: 5px;
      margin: 10px;
      line-height: 10px;
      position: unset !important;
      z-index: 1;
      color: rgb(255, 254, 254);
      text-align: center;
    }
  }
  .details-list {
    display: flex;
    flex-direction: column;
    width: 100%;
    height: 100%;
    align-self: center;
    background: none;
    .details + .details {
      margin-top: 5px;
    }
  }
}
