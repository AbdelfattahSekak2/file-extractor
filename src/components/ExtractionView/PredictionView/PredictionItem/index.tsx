import * as React from "react";

import "./index.scss";

interface Element {
  value: string | number;
  sign?: string;
  subValue?: {
    label: string;
    value: string | number;
  }
  probability: number;
}

interface IProps {
  label: string;
  element: Element | Element[];
}

export const getProbability = (probability: number) => ({
    width: `${probability * 100}%`,
    borderRadius: probability === 1 ? "0 6px 6px 0" : "0",
  })

const SubValue = ({ value, label }: { value: string | number; label: string }) => <>
  ({label} : <strong>{value}</strong>)
</>;

const PredictionItem: React.FunctionComponent<IProps> = ({
  label,
  element,
}) => {
  return <div className="prediction-item">
    <label>{label}</label>
    {
      Array.isArray(element)
        ? <div className="details-list">
          {
            element.map((item: Element, key: number) =>
              <div className="details" key={key} >
                <p>
                  {item.value}{item.sign} {item.subValue ? <SubValue {...item.subValue} /> : null}
                </p>
                <div className="bar" style={getProbability(item.probability)} />
              </div>
            )
          }
        </div>
        : <div className="details">
          <p>{element.value}{element.sign}</p>
          <div className="bar" style={getProbability(element.probability)} />
        </div>
    }
  </div>;
};

export default PredictionItem;

