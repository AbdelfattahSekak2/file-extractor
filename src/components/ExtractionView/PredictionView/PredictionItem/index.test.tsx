import React from "react";
import { ShallowWrapper, shallow } from "enzyme";

import PredictionItem, { getProbability } from "."

describe("PredictionItem component", () => {
  const element = {
    value: "value",
    probability: 1,
  };
  const defaultProps = {
    label: "label",
    element
  }
  let wrapper: ShallowWrapper;
  it("should render correctly", () => {
    wrapper = shallow(<PredictionItem {...defaultProps} />);
    expect(wrapper).not.toBeNull();
    expect(wrapper).toMatchSnapshot();
  });

  it("should should match snapshot when element is a list", () => {
    wrapper = shallow(<PredictionItem {...defaultProps} element={[element, element]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should should match snapshot when subValue is defined", () => {
    wrapper = shallow(<PredictionItem {...defaultProps} element={[
      {
        ...element,
        subValue: {
          value: "value",
          label: "label"
        },
      }
      , element]} />);
    expect(wrapper).toMatchSnapshot();
  });
});


describe("getProbability function", () => {

  it("should return the correct object when the probability is other than 1", () => {
    expect(getProbability(0.8)).toEqual({
      width: "80%",
      borderRadius: "0",
    });
  });
  it("should return the correct object when the probability equal 1", () => {
    expect(getProbability(1)).toEqual({
      width: "100%",
      borderRadius: "0 6px 6px 0",
    });
  });
});
