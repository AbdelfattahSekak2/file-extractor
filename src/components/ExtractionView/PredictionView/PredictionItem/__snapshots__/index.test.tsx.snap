// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PredictionItem component should render correctly 1`] = `
<div
  className="prediction-item"
>
  <label>
    label
  </label>
  <div
    className="details"
  >
    <p>
      value
    </p>
    <div
      className="bar"
      style={
        Object {
          "borderRadius": "0 6px 6px 0",
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;

exports[`PredictionItem component should should match snapshot when element is a list 1`] = `
<div
  className="prediction-item"
>
  <label>
    label
  </label>
  <div
    className="details-list"
  >
    <div
      className="details"
      key="0"
    >
      <p>
        value
         
      </p>
      <div
        className="bar"
        style={
          Object {
            "borderRadius": "0 6px 6px 0",
            "width": "100%",
          }
        }
      />
    </div>
    <div
      className="details"
      key="1"
    >
      <p>
        value
         
      </p>
      <div
        className="bar"
        style={
          Object {
            "borderRadius": "0 6px 6px 0",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`PredictionItem component should should match snapshot when subValue is defined 1`] = `
<div
  className="prediction-item"
>
  <label>
    label
  </label>
  <div
    className="details-list"
  >
    <div
      className="details"
      key="0"
    >
      <p>
        value
         
        <SubValue
          label="label"
          value="value"
        />
      </p>
      <div
        className="bar"
        style={
          Object {
            "borderRadius": "0 6px 6px 0",
            "width": "100%",
          }
        }
      />
    </div>
    <div
      className="details"
      key="1"
    >
      <p>
        value
         
      </p>
      <div
        className="bar"
        style={
          Object {
            "borderRadius": "0 6px 6px 0",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;
